src/virt.rs

```diff
@@ -562,11 +558,19 @@ impl VirtContext {
     }
 
     /// Handle the trap coming from the firmware
-    pub fn handle_firmware_trap(&mut self, mctx: &mut MiralisContext) {
+    pub fn handle_firmware_trap(&mut self, mctx: &mut MiralisContext, policy: &mut Policy) {
         let cause = self.trap_info.get_cause();
         match cause {
-            MCause::EcallFromSMode if Policy::ecall_from_firmware(self).overwrites() => {
+            MCause::EcallFromSMode if policy.ecall_from_payload(mctx, self).overwrites() => {
```

Comment:
Is it `ecall_from_payload` or for firmware? I guess here we should never get an `EcallFromSMode` anyway as we come from the firmware which is in U-mode.

Reply:
True, mistake from my side